internal/generator/generator_util.go

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package generator
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// regex.
+const (
+	// Extract single directive definition block
+	// static ngx_command_t  {name}[] = {definition}
+	// this regex extracts {name} and {definition}.
+	extractDirectivesDefBlock = "ngx_command_t\\s+(\\w+)\\[\\]\\s*=\\s*{(.*?)};"
+
+	// Extract one directive definition and attributes from extracted block
+	// { ngx_string({directive_name}),
+	//   {bitmask1|bitmask2|...},
+	//   ... },
+	// this regex extracts {directive_name} and {bitmask1|bitmask2|...}.
+	extractSingleDirective = "ngx_string\\(\"(.*?)\"\\).*?,(.*?),"
+
+	extractSingleLineComment = `//.*`
+
+	extractMultiLineComment = `/\*[\s\S]*?\*/`
```

Comment:
Nit: Unless you need to reference these strings in a few different places, I think it's better to move this down the the call to `regexp.MustCompile`

```go
var (
	directivesDefBlockExtracter = regexp.MustCompile("ngx_command_t\\s+(\\w+)\\[\\]\\s*=\\s*{(.*?)};")
	// ...rest
)
```

De-duplicating string constants can be a good idea, but I think when applied too aggressively it hurts readability.